pkg/apis/options/providers.go

```diff
@@ -255,3 +262,38 @@ func providerDefaults() Providers {
 	}
 	return providers
 }
+
+// enrichWithDefaults enriches all Providers' config with values from the
+// single default
+func (providers Providers) enrichWithDefaults() Providers {
+	var enrichedProviders []Provider
+
+	for _, provider := range providers {
+		var (
+			enrichedProvider              Provider
+			newGenericMap, provGenericMap map[string]interface{}
+		)
+
+		provJSON, _ := json.Marshal(provider)
+		if err := json.Unmarshal(provJSON, &provGenericMap); err != nil {
+			println(err)
+		}
+
+		defJSON, _ := json.Marshal(providerDefaults()[0])
```

Comment:
Rather than using `providerDefaults`, have you considered adding a `default()` function to provider which then checks the existence of fields and defaults them? I had intended long term to remove the `providerDefaults` and equivalent functions by doing something like I've just described

I'm imagining long term that every struct in the package will have a `default()` function that will then be called by the parent `default()` function, until we get to the top level which will have a public `Default()`

There is also a possible argument for making them all public (so the defaults can be used in other tests throughout the code, but maybe we can leave that for now)